diff --git a/lib/composables/router.js b/lib/composables/router.js
index db5aad75ec9681105794f99e0e6ed093cf4c594c..89085628576755c2641578b5ca3414e2594547df 100644
--- a/lib/composables/router.js
+++ b/lib/composables/router.js
@@ -32,7 +32,17 @@ export function useLink(props, attrs) {
   // vue-router useLink `to` prop needs to be reactive and useLink will crash if undefined
   const routerLink = RouterLink.useLink({
     to: toRef(() => props.to || ''),
-    replace: toRef(() => props.replace)
+    replace: toRef(() => props.replace),
+    prefetch: computed(() => {
+      if (props.prefetch !== undefined && props.prefetch !== null) {
+        return props.prefetch;
+      }
+      if (attrs.prefetch !== undefined && attrs.prefetch !== null) {
+        return attrs.prefetch;
+      }
+      return false;
+    }),
+    cacheFor: computed(() => props.cacheFor ?? attrs.cacheFor),
   });
   // Actual link needs to be undefined when to prop is not used
   const link = computed(() => props.to ? routerLink : undefined);
@@ -57,7 +67,11 @@ export function useLink(props, attrs) {
       href,
       'aria-current': toRef(() => isActive.value ? 'page' : undefined),
       'aria-disabled': toRef(() => props.disabled && isLink.value ? 'true' : undefined),
-      tabindex: toRef(() => props.disabled && isLink.value ? '-1' : undefined)
+      tabindex: toRef(() => props.disabled && isLink.value ? '-1' : undefined),
+      onMouseenter: toRef(() => routerLink.onMouseenter),
+      onMouseleave: toRef(() => routerLink.onMouseleave),
+      onMousedown: toRef(() => routerLink.onMousedown),
+      onMouseup: toRef(() => routerLink.onMouseup)
     })
   };
 }
